apiserver/facades/client/modelmanager/modelmanager.go

```diff
@@ -806,55 +806,71 @@ func (m *ModelManagerAPI) ListModelSummaries(ctx context.Context, req params.Mod
 		return result, errors.Trace(err)
 	}
 
-	modelInfos, err := m.state.ModelSummariesForUser(userTag, req.All && m.isAdmin)
-	if err != nil {
-		return result, errors.Trace(err)
+	var modelInfos []model.ModelSummary
+	if req.All {
+		if m.isAdmin {
+			modelInfos, err = m.modelService.ListAllModelSummaries(ctx)
+			if err != nil {
+				return result, errors.Trace(err)
+			}
+		} else {
+			return result, fmt.Errorf("%w: cannot list all models as non-admin user", apiservererrors.ErrPerm)
+		}
+	} else {
+		modelInfos, err = m.modelService.ListModelSummariesForUser(ctx, userTag.Id())
+		if err != nil {
+			return result, errors.Trace(err)
+		}
 	}
```

Comment:
This is an if soup, can we clean it up. Otherwise, the PR looks good.

Reply:
Sorted it out a bit